ear-production-suite-plugins/lib/include/communication/metadata_thread.hpp

```diff
@@ -0,0 +1,75 @@
+//
+// Created by Richard Bailey on 06/04/2022.
+//
+
+#ifndef EAR_PRODUCTION_SUITE_METADATA_THREAD_HPP
+#define EAR_PRODUCTION_SUITE_METADATA_THREAD_HPP
+#include <thread>
+#include <mutex>
+#include <condition_variable>
+#include <vector>
+#include <functional>
+#include <atomic>
+
+namespace ear::plugin {
+class MetadataThread {
+public:
+    using MessageQueue = std::vector<std::function<void()>>;
+    MetadataThread() : thread_([this](){updateLoop();}) {
```

Comment:
Does this set up a race condition? Can we guarantee messages_ is initialised before the thread calls updateLoop (which accesses messages_ )

Reply:
Good catch, yeah I think we need to move the messages_ declaration above the thread_ declaration.